File: src/main/java/tech/zolhungaj/amqapi/servercommands/globalstate/RankedLeaderboardUpdate.java
package tech.zolhungaj.amqapi.servercommands.globalstate;

import com.squareup.moshi.Json;
import tech.zolhungaj.amqapi.servercommands.Command;
import tech.zolhungaj.amqapi.servercommands.CommandType;
import tech.zolhungaj.amqapi.servercommands.objects.RankedLeaderboardEntry;
import tech.zolhungaj.amqapi.constants.AmqRanked;

import java.util.List;

public record RankedLeaderboardUpdate(
        @Json(name = "serieId")AmqRanked.RankedSeries rankedSeries,
        @Json(name = "standings") List<RankedLeaderboardEntry> leaderboard
)implements Command {

    @Override
    public String commandName() {
        return CommandType.RANKED_LEADERBOARD_UPDATE.commandName;
    }
}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/globalstate/RankedLeaderboardUpdate.kt
package tech.zolhungaj.amqapi.servercommands.globalstate

import com.squareup.moshi.Json
import tech.zolhungaj.amqapi.constants.RankedSeries
import tech.zolhungaj.amqapi.servercommands.CommandType
import tech.zolhungaj.amqapi.servercommands.objects.RankedLeaderboardEntry

@JvmRecord
@CommandType("ranked standing updated")
data class RankedLeaderboardUpdate(
    @Json(name = "serieId") val rankedSeries: RankedSeries,
    @Json(name = "standings") val leaderboard: List<RankedLeaderboardEntry>
)
